### PR TITLE
HDDS-10591. [hsync] improve block token refresh message.

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?
HDDS-10591. [hsync] improve block token refresh message.

Please describe your PR in detail:
Usually an expired block token received by a DataNode is refreshed properly. However the relevant log messages look scary and confusing.

* Some of the logs should tone down. Instead of ERROR, it should be a WARN, because the exception will be retried.
* The log messages surrounding block token refresh should be more clear.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10591

## How was this patch tested?

Log message only. Existing unit tests should suffice.

### CODE REVIEW

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java [hunk 1]
```diff
@@ -217,18 +217,19 @@ private boolean isConnectivityIssue(IOException ex) {
...
       if (blockLocationInfo == null) {
         LOG.debug("No new block location info for block {}", blockID);
       } else {
-        LOG.debug("New pipeline for block {}: {}", blockID,
-            blockLocationInfo.getPipeline());
         setPipeline(blockLocationInfo.getPipeline());
         tokenRef.set(blockLocationInfo.getToken());
+        LOG.info("New pipeline for block {}: {}", blockID,
+            blockLocationInfo.getPipeline());
+        LOG.info("A new token is added {}", blockLocationInfo.getToken());
```
Comment (ChenSammi) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:232]: It's better not log the token as it's a sensitive information.

Comment (jojochuang) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java]: How about logging the expiry time of the token?

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java [hunk 2]
```diff
@@ -217,18 +218,27 @@ private boolean isConnectivityIssue(IOException ex) {
...
+            blockLocationInfo.getPipeline());
+
+        if (blockLocationInfo.getToken() != null) {
+          tokenRef.set(blockLocationInfo.getToken());
+          OzoneBlockTokenIdentifier tokenId = new OzoneBlockTokenIdentifier();
+          tokenId.readFromByteArray(tokenRef.get().getIdentifier());
+          LOG.info("A new token is added for block {}. Expiry: {}",
+              blockID, Instant.ofEpochMilli(tokenId.getExpiryDate()));
+        } else {
+          LOG.warn("Unable to add a new block token for block {}", blockID);
```
Comment (ChenSammi) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:240]: If security is not enabled, then token will be null, this WARN log will be confusing, I would suggest remove it.

Comment (jojochuang) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:240]: Oh yeah you're right. Good catch.

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java [hunk 3]
```diff
@@ -217,18 +218,25 @@ private boolean isConnectivityIssue(IOException ex) {
...
       } else {
-        LOG.debug("New pipeline for block {}: {}", blockID,
-            blockLocationInfo.getPipeline());
         setPipeline(blockLocationInfo.getPipeline());
-        tokenRef.set(blockLocationInfo.getToken());
+        LOG.info("New pipeline for block {}: {}", blockID,
+            blockLocationInfo.getPipeline());
+
+        if (blockLocationInfo.getToken() != null) {
+          tokenRef.set(blockLocationInfo.getToken());
```
Comment (smengcl) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:234]: When would `blockLocationInfo.getToken()` be null inside `refreshBlockInfo()` call here?

Comment (jojochuang) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:234]: possible in unsecure environment where block token is disabled.

Comment (jojochuang) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java:234]: I'm fine either way. blockLocationInfo.getToken() should only be null if block token is disabled. In which case, tokenRef would have been null too already.

**Review by smengcl — APPROVED** [review]
+1. Thanks @jojochuang for the patch!

**Comment by jojochuang** [comment]
Merged. Thanks Siyao and Sammi

**Pull request merged**